### PR TITLE
Add Map Flavours with FDroid flavor and Google Maps flavor

### PR DESCRIPTION
**Pull request by creativepsyco**
Added gradle flavors for FDroid and Google Play store. 

To work within a particular flavor switch the variant to either `fdroidDebug/Release` or `googleplayDebug/Release` and then use run from the IDE. See screenshot

![Image of IDE][1]

From commandline
```bash
./gradlew :app:assembleGoogleplayDebug
# Or
./gradlew :app:assembleFdroidDebug
# The command below will generate both APKs (debug) in the `app/build/outputs/apk/` folder
./gradlew :app:assembleDebug
```
Also read [Gradle Plugin Documentation][2] regarding product flavors and watch the video about the [Gradle Flavor stuff][3]

Each Flavor needs to implement `MapModuleFactory` providing a `MapModule` which can do stuff as decided by the interface for `MapModule`. Right now I just made it to return me a Fragment, but you can do other stuff as well.

![Image of IDE][4]

[1]: https://www.dropbox.com/s/hqwuak5cew4vg17/Screenshot%202015-06-13%2002.13.15.png?raw=1
[2]: http://tools.android.com/tech-docs/new-build-system/user-guide#TOC-Product-flavors
[3]: https://www.youtube.com/watch?v=LCJAgPkpmR0
[4]: https://www.dropbox.com/s/5v7z7ghvzpvkla8/Screenshot%202015-06-13%2002.22.18.png?raw=1

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -46,4 +53,10 @@ dependencies {
     compile 'com.android.support:design:22.2.0'
     compile 'com.squareup.picasso:picasso:2.5.2'
 
+    googleplayCompile 'com.google.android.gms:play-services-base:7.5.0'
+    googleplayCompile 'com.google.android.gms:play-services-maps:7.5.0'
+
```
Comment (mananwason) [inline app/build.gradle]: I don't think that you need to include base here

File: app/src/fdroid/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <!-- These are the permissions required by FDroid -->
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
+    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
+    <uses-permission android:name="android.permission.INTERNET"/>
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
+
```
Comment (mananwason) [inline app/src/fdroid/AndroidManifest.xml]: I don't think that we need the external storage.

**Comment by mananwason** [comment]
@creativepsyco Please checkout my comments, then we can merge this :smile:. Nice work by the way. This flavours thing is really cool.

File: app/src/fdroid/AndroidManifest.xml (hunk 2 above)
Comment (creativepsyco) [inline app/src/fdroid/AndroidManifest.xml]: See https://code.google.com/p/osmdroid/wiki/Prerequisites

**Comment by creativepsyco** [comment]
Updated.

File: app/src/fdroid/AndroidManifest.xml (hunk 2 above)
Comment (mananwason) [inline app/src/fdroid/AndroidManifest.xml]: okay Apparently it needs to download some data.

**Comment by mananwason** [comment]
Cool but it says Travis failed. I can't make out why. Kindly check.

**Comment by creativepsyco** [comment]
Just a timing issue on Travis. sometimes the ADB fails to upload the APK.

**Comment by mananwason** [comment]
How did you tell?

**Comment by creativepsyco** [comment]
Here is the error:

```bash
com.android.ddmlib.ShellCommandUnresponsiveException
	at com.android.ddmlib.AdbHelper.executeRemoteCommand(AdbHelper.java:513)
	at com.android.ddmlib.Device.uploadAPK(Device.java:1015)
	at com.android.ddmlib.Device.installPackages(Device.java:895)
	at com.android.builder.testing.ConnectedDevice.installPackages(ConnectedDevice.java:105)
	at com.android.builder.internal.testing.SimpleTestCallable.call(SimpleTestCallable.java:125)
	at com.android.builder.internal.testing.SimpleTestCallable.call(SimpleTestCallable.java:48)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Unable to install /home/travis/build/fossasia/open-event-android/app/build/outputs/apk/app-debug.apk
com.android.ddmlib.InstallException: Unable to upload some APKs
	at com.android.ddmlib.Device.installPackages(Device.java:913)
	at com.android.builder.testing.ConnectedDevice.installPackages(ConnectedDevice.java:105)
	at com.android.builder.internal.testing.SimpleTestCallable.call(SimpleTestCallable.java:125)
	at com.android.builder.internal.testing.SimpleTestCallable.call(SimpleTestCallable.java:48)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
com.android.builder.testing.ConnectedDevice > runTests[test(AVD) - 5.0] FAILED 
	com.android.builder.testing.api.DeviceException: com.android.ddmlib.InstallException: Unable to upload some APKs
		at com.android.builder.testing.ConnectedDevice.installPackages(ConnectedDevice.java:108)
null
```

Here's the solution:

http://stackoverflow.com/questions/28949722/android-tests-fail-on-travis-with-shellcommandunresponsiveexception

But sometimes doesn't work for some reason.

**Comment by mananwason** [comment]
Cool It has passed. So I'll merge it.

**Pull request merged**